lib/twitter_cldr/utils/script_detector.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+module TwitterCldr
+  module Utils
+    class ScriptDetectionResult
+
+      attr_reader :scores
+
+      def initialize(scores)
+        @scores = scores
+      end
+
+      def best_guess
+        max_score = 0
+        max_script_name = nil
+
+        scores.each_pair do |script_name, score|
```

Comment:
Can we use [Enumerable#max_by](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-max_by) here?